grype/logger/logger.go

```diff
@@ -9,4 +9,5 @@ type Logger interface {
 	Info(args ...interface{})
 	Debugf(format string, args ...interface{})
 	Debug(args ...interface{})
+	IsDebugEnabled() bool
```

Comment:
having these types of flags is typical in log4j :trollface:

Reply:
Tagging onto https://github.com/anchore/grype/pull/559/files#r774058215 , it would be ideal to not expand the logging interface unless we really need to (say to add another logging level). Are we certain we need this information? The larger we make this interface the harder it is to plug in other loggers or have other projects interop with the logger here (we even held off for a long time before adding the warn log level to the interface!).